best-practices/README.md

```diff
@@ -104,11 +104,18 @@ Rails
 Testing
 -------
 
+(See also the rspec style guide that appears in this section)
+
+* First #describe what you are doing.
```

Comment:
I found it really helpful, when pairing with @novemberkilo, to see him basically write out the whole structure for a spec first, including descriptions, before writing any executable spec code.

Something like:
```
describe Foo do
  describe '#bar' do
    it 'does thing a'
    it 'does thing b'
    context 'with baz' do
      it 'does thing c'
    end
  end
end
```

Any problems with the logical organisation of your specs are much easier to pick up early if you do that than if you just start writing the first spec, get it passing, then tack on another etc. Maybe we should recommend that approach.